src/helics/common/timeRepresentation.hpp

```diff
@@ -105,7 +105,8 @@ class integer_time
         auto divBase = static_cast<base> (div);
         double frac = div - static_cast<double> (divBase);
         baseType nseconds = (divBase << N) + static_cast<base> (frac * multiplier);
-        return (t < -1e12) ? ((t > 1e12) ? nseconds : maxVal ()) : minVal ();
+        return (t < toDouble (minVal () + 10000)) ? ((t > toDouble (maxVal () - 10000)) ? nseconds : maxVal ()) :
```

Comment:
Why was 10000 chosen?

Reply:
The maxTime in C++ has integral representation of 9223372036854775807,  the double representation of this gets converted back to 9223372036854774784.  This means a difference of 1024  is the lowest that number could be used to make this consistent.  10000 is 10 microseconds at the default time resolution which seems close enough to not matter that much and give a little more wiggle room.  I think it has to be done at the underlying representation level to be consistent as well.  

The other option was the float representation of Time::maxTIme()  but that would have been ~26 seconds worth of time shift,  that seemed to be kind of a waste since I don't think that would come into play that often if ever.

Reply:
I should put a note in the code about why it is there though.